whetstone/README.md

```diff
@@ -0,0 +1,278 @@
+for Fragments inside the generated code.
+
+The last part of the integration is that Whetstone will automatically generate
+a `NavDestination` for the screen by using the `route` and `type`. This
+generated destination is automatically provided into a `Set` in the component
+that uses `destinationScope` as it's scope (usually an app wide or Activity
+level scope). With that it's not necessary to manually create a `Set`
+of all destinations anymore. It can simply be injected.
+
+
+## Example
+
+This is a minimal example of how using Whetstone for a screen with the Navigator integration 
+would look like.
+
+```kotlin
+// marker class for the scope
+sealed interface Example
+
+// dependencies interface that is contributed to the AppScope component
+@ContributesTo(AppScope::class)
+interface ExampleDependencies {
+    val repository: ExampleRepository // makes ExampleRepository from AppScope available
+}
+
+// state machine survives orientation changes
+@ScopeTo(Example::class)
+internal class ExampleStateMachine @Inject constructor(
+    val route: ExampleRoute, // inject the navigator route that was used to get to this screen
+    val repository: ExampleRepository, // the repository
+    val navigator: ExampleNavigator,
+) : StateMachine { ... }
+
+// scope the navigator so that everything interacts with the same instance
+@ScopeTo(Example::class) 
+// make ExampleNavigator available as NavEventNavigator so that the generated code can automatically
+// set up the navigation handling
+@ContributesBinding(Example::class, NavEventNavigator::class) 
+class ExampleNavigator @Inject constructor() : NavEventNavigator() { ... }
+
+@ComposeScreen(
+    scope = Example::class, // uses our marker class
+    parentScope = AppScope::class, // same parent scope as we use to contribute ExampleDependencies
+    dependencies = ExampleDependencies::class, // the dependencies interface
+    stateMachine = ExampleStateMachine::class, // the state machine used for this ui
+)
+@NavDestination(
+    route = ExampleRoute::class, // the route used to navigate to ExampleUi
+    type = DestinationType.SCREEN, // whether it's a screen, dialog or bottom sheet
+    destinationScope = AppScope::class, // contribute the generated destination to AppScope
+)
+@Composable
+internal fun ExampleUi(
+    state: ExampleState,
+    sendAction: (ExampleAction) -> Unit,
+) { 
+    // render the ui for ExampleState
+}
+```
+
+Using this would require a one time setup in the app so that the screens can look up the `AppScope`
+component through `getSystemService` to retrieve the dependencies:
+
+```kotlin
+@Singleton
+@MergeComponent(scope = AppScope::class)
+interface AppComponent {
+    // allows an Activity to get all generated NavDestinations to set up the NavHost
+    val destinations: Set<NavDestination>
+    
+    @Component.Factory
+    interface Factory {
+        fun create(): AppComponent
+    }
+}
+
+class App : Application(), ImageLoaderFactory {
```

Comment:
Small copy/paste error: The `ImageLoaderFactory` is not part of the example

Reply:
```suggestion
class App : Application() {
```